client/ayon_silhouette/plugins/load/load_source.py

```diff
@@ -30,28 +44,43 @@ def load(self, context, name=None, namespace=None, options=None):
 
         filepath = self.filepath_from_context(context)
 
-        # Add Source item to the project
-        source = fx.Source(filepath)
-
-        # Provide a nice label indicating the product
-        source.label = self._get_label(context)
-        project.addItem(source)
-
-        # property.hidden = True  # hide the attribute
-        lib.imprint(source, data={
-            "name": str(name),
-            "namespace": str(namespace),
-            "loader": str(self.__class__.__name__),
-            "representation": context["representation"]["id"],
-        })
-
-        # container = pipeline.containerise(
-        #     name=str(name),
-        #     namespace=str(namespace),
-        #     nodes=nodes,
-        #     context=context,
-        #     loader=str(self.__class__.__name__),
-        # )
+        # A source file may contain multiple parts, such as a left view
+        # and a right view in a single EXR.
+        options = options or {}
+        load_all_parts = options.get("load_all_parts", True)
+
+        has_multiple_parts = os.path.splitext(filepath)[-1].lower() in {
+            ".exr", ".sxr"
+        } and fx.Source(filepath, part=1).video
```

Comment:
No idea whether only EXR/SXR can be multiple parts, but on a h264 video it seems no matter what `part=n` number I put in Silhouette always reports it has video and it's just the same video - which makes it seems like `part=` does not do anything on those other formats.

There must be a beter way to identify whether it has multiple parts to load.
See: https://forum.borisfx.com/t/python-load-both-channels-of-stereo-exr-sxr/20215/3?u=bigroy

@philippe-ynput @jakubjezek001 any ideas?

Reply:
For now I assumed only exr/sxr - and identifier subimages via a call to `ayon_core.lib.transcoding.get_oiio_info_for_input` with `subimages=True`. Seems to work.